docs/queryworkers/attribute-based-access.md

```diff
@@ -0,0 +1,260 @@
+}
+```
+
+On the other hand, if Partner 1 tries to run the same command to edit data belonging to Partner 2:
+
+```
+curl -X 'POST' \\
+    '<https://fulfillment.eng.macrometa.io/_fabric/_system/_api/restql/execute/PartnerOrdersUpdate>' \\
+    -H 'accept: application/json' \\
+    -H 'Content-Type: application/json' \\
+    -H 'Authorization: apikey <PARTNER 1 API KEY>' \\
+    -d '{ "bindVars": { "customer": 2}}'
+```
+
+The result is empty because the attempt failed:
+
+```sql
+{
+   {}
+}
+```
+
+## Example 3
+
+In this example, partners attempt to view employee data and are blocked because they do not have proper access.
+
+The name of this query worker is `Employees`:
+
+```sql
+FOR doc IN employees
+    FILTER CURRENT_APIKEY_ATTRIBUTE("partner") == null
+    RETURN doc
+```
+
+The result is empty because API keys with the `partner` attribute do not have access to the `employees` collection:
+
+```sql
+{
+   {}
+}
+```
+
+## Example 4
+
+In this example, employees can see all items and orders but cannot update them, and partners can view and update their own items.
+
+The name of this query worker is `ViewOrders`.
+
+```sql
+FOR doc IN orders
+    FILTER doc.partner == CURRENT_APIKEY_ATTRIBUTE("partner") OR 
+            CURRENT_USER_ATTRIBUTE("employee") == "admin" OR
+            CURRENT_USER_ATTRIBUTE("employee") == "staff"
+    RETURN doc
+```
+
+Partner 1 can run a command to add data to their order:
+
+```
+curl -X 'POST' \\
+    '<https://fulfillment.eng.macrometa.io/_fabric/_system/_api/restql/execute/ViewOrders>' \\
+    -H 'accept: application/json' \\
+    -H 'Content-Type: application/json' \\
+    -H 'Authorization: apikey <PARTNER 1 API KEY>' \\
+    -d '{ "bindVars": {}}'
+```
+
+The result shows the added data:
+
+```sql
+{
+   { "customer": 1, "item": "hammer", "ordered": 2, "partner": "partner1" }
+}
+```
+
+Similarly, if an employee is logged onto their account they can run a command to view all data:
+
+```
+curl -X 'POST' \\
+    '<https://fulfillment.eng.macrometa.io/_fabric/_system/_api/restql/execute/ViewOrders>' \\
+    -H 'accept: application/json' \\
+    -H 'Content-Type: application/json' \\
+    -H 'Authorization: bearer eyJhbGciOiJIUzI1NiIsInR5cC...' \\
+    -d '{ "bindVars": {}}'
+```
+
+The result shows data for both customers since the employee has view access to both.
+
+```sql
+{
+   { "customer": 1, "item": "hammer", "ordered": 2, "partner": "partner1" },
+   { "customer": 2, "item": "drill", "ordered": 1, "partner": "partner2" }
+}
+```
+
+
+The name of this Query Worker is `OrdersUpdate`.
+
+```sql
+FOR doc IN orders
+    FILTER doc.customer == @customer AND
+            doc.partner == CURRENT_APIKEY_ATTRIBUTE("partner") OR
+            CURRENT_USER_ATTRIBUTE("employee") == "admin"
+    UPDATE doc WITH {"ordered": doc.ordered + 1} IN orders
+    RETURN NEW
+```
+
+Partner 1 can run a command to add data to their order:
+
+```
+curl -X 'POST' \\
```

Comment:
Still not seeing added data.

Reply:
Unless the 1 corresponds to hammer? Where is it saying that they ordered three hammers?

Reply:
@diana-macrometa Worked with Joubert to fix the example descriptions.